cmake/Modules/FindSYCL.cmake

```diff
@@ -0,0 +1,500 @@
+#.rst:
+# FindSYCL
+# --------
+#
+# .. note::
+
+# The following variables affect the behavior of the macros in the script needed
+# to be defined before calling ``SYCL_ADD_EXECUTABLE`` or ``SYCL_ADD_LIBRARY``::
+#
+#  SYCL_COMPILER
+#  -- SYCL compiler's executable.
+#
+#  SYCL_FLAGS
+#  -- SYCL compiler's compilation command line arguments.
+#
+#  SYCL_HOST_FLAGS
+#  -- SYCL compiler's 3rd party host compiler (e.g. gcc) arguments .
+#
+#  SYCL_INCLUDE_DIR
+#  -- Include directory for SYCL compiler/runtime headers.
+#
+#  SYCL_LIBRARY_DIR
+#  -- Include directory for SYCL compiler/runtime libraries.
+
+# Helpers::
+# Introduce SYCL compiler to build .cpp containing SYCL kernel.
+#
+#  SYCL_ADD_EXECUTABLE
+#
+#  SYCL_ADD_LIBRARY
+
+macro(SYCL_FIND_HELPER_FILE _name _extension)
+  set(_full_name "${_name}.${_extension}")
+  # CMAKE_CURRENT_LIST_FILE contains the full path to the file currently being
+  # processed.  Using this variable, we can pull out the current path, and
+  # provide a way to get access to the other files we need local to here.
+  get_filename_component(CMAKE_CURRENT_LIST_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+  set(SYCL_${_name} "${CMAKE_CURRENT_LIST_DIR}/FindSYCL/${_full_name}")
+  if(NOT EXISTS "${SYCL_${_name}}")
+    set(error_message "${_full_name} not found in ${CMAKE_CURRENT_LIST_DIR}/FindSYCL")
+    message(FATAL_ERROR "${error_message}")
+  endif()
+  # Set this variable as internal, so the user isn't bugged with it.
+  set(SYCL_${_name} ${SYCL_${_name}} CACHE INTERNAL "Location of ${_full_name}" FORCE)
```

Comment:
"CACHE" means this value cannot be changed even in run-to-run case, until you clean previous build caches.
Is it what we want?

Reply:
All CACHE decorated variables here indicate files which should not be changed, like sycl compiler executable, run_sycl.cmake, and compilation dependent files.